architecture/0007-use-weblate-for-i18n.md

```diff
@@ -0,0 +1,27 @@
+# 7. Use weblate for i18n
+
+Date: 2020-05-07
+
+## Status
+
+Accepted
+
+## Context
+
+We have a request to provide a french translationed version of de-africa. We are using the i18next and i18next-react libraries with all our UI strings within en/translation.json. There are also two long life branches master & next representing our current production release (v7) and the next production release (v8). Due to the time lag between releases, there are hundreds more strings in the v7 translation.json versus the master version.
+
+It would not be workable or scalable for translators to provide these translations into the github repository directly as this requires a level of technical expertise and we won't have an easy way of keeping track of which strings haven't been translated yet. We also require some quality check mechanism to ensure the quality of the contributed translations.
+
+## Decision
+
+We will use hosted weblate at https://hosted.weblate.org/projects/terriajs/ to allow external translators to contribute to the project. There will be two projects, one connected to the master branch and another connected to the next branch.
+
+When the en/translation.json is updated, our weblate plugin and associated webhook will let the hosted weblate know and weblate will pull the updated changes. When a contributing translator provides translations, weblate will create a PR into the corresponding branch allowing a terria mainter to review the changes before merging the changes.
```

Comment:
Would pull requests with whole translation files for other localisations fit into this approach e.g. https://github.com/TerriaJS/terriajs/pull/5241 & https://github.com/TerriaJS/terriajs/pull/5302?

I'm not saying it should. Maybe Weblate has something that could help us get these efforts into the project through the Weblate workflow below.

Reply:
I see that weblate has a mechanism to review translations. I am not sure who is in charge of reviewing, but I wonder when we get entire terriajs translated to French and approve those translations. When some source string changes, will it set its translation to unreviewed or need editing status?

Reply:
@steve9164 New whole translation files won't be an issue as it'll get picked up by the github webhook and then it'll follow the existing weblate process. If it was an existing translation file however, this might cause merge conflicts between our respository and the weblate repository.
@zoran995 - I believe that if the source string changes, it will change the status of the translated string to 'needs editing'.

Reply:
Just checked above with weblate support to be sure. When source strings update, their state will change automatically to "Needs editing", and if we are using reviewing new review will be required. Can we add that somewhere to adr, and also, should reviewing be activated?

Reply:
Keep in mind that we will need additional work to support RTL languages currently

Reply:
I don't think we should enable reviewing at this stage, maybe once we have identified someone who could do it? Thoughts? @steve9164